guides/tutorials/codelabs/pubsub-to-appengine/index.md

```diff
@@ -0,0 +1,133 @@
+---
+layout: single
+title: "Codelab: Deploying GCS Pubsub Artifacts to Appengine"
+sidebar:
+  nav: guides
+---
+
+{% include toc %}
+
+In this codelab, you will deploy artifacts to Google App Engine (GAE) via a Spinnaker pipeline
+configured to trigger on pubsub messages from a GCS bucket serving as an artifact repository.
+The pubsub messages from the GCS bucket will contain contextual information that will allow
+Spinnaker to parse and deploy the stored artifacts.
+
+This codelab is estimated to take an hour.
+
+## Overview
+
+The end goal of this codelab is to set up an environment where a GCS upload of a tarball
+(containing a GAE application) triggers a pipeline that will deploy that application to
+GAE. The mechanism we'll use to inform the pipeline about the application is a pubsub message
+sent from GCS and ingested by Spinnaker.
+
+## Prerequisites
+
+This codelab assumes you have a billing-enabled GCP project.
+
+## Set Up Your Environment
+
+### Create a GCS Bucket to store Artifacts
+
+In the environment where you have `gcloud` installed, run the following commands:
+
+```
+$ export BUCKET_NAME=gs://<some_name> && export PROJECT_ID=<project_id>
+$ gcloud auth login
+$ gsutil mb -p $PROJECT_ID $BUCKET_NAME
+```
+
+### Set up Google Cloud Pubsub to listen to bucket object changes
+
+* The steps to configure GCS to publish pubsub messages are laid out [in the docs](https://cloud.google.com/storage/docs/reporting-changes), but we'll summarize the steps here.
```

Comment:
For consistency with the rest of the docs I would drop the bullets at the start of each section